src/consumption.cpp

```diff
@@ -252,80 +246,83 @@ edible_rating player::can_eat( const item &food, std::string *err ) const
                          ? has_charges( comest->tool, 1 )
                          : has_amount( comest->tool, 1 );
         if( !has ) {
-            return error( NO_TOOL, string_format( _( "You need a %s to consume that!" ),
-                                                  item::nname( comest->tool ).c_str() ) );
+            return edible_ret_val::failure( string_format( _( "You need a %s to consume that!" ),
+                                            item::nname( comest->tool ).c_str() ), NO_TOOL );
         }
     }
 
     // For all those folks who loved eating marloss berries.  D:< mwuhahaha
     if( has_trait( trait_id( "M_DEPENDENT" ) ) && food.typeId() != "mycus_fruit" ) {
-        return error( INEDIBLE_MUTATION, _( "We can't eat that.  It's not right for us." ) );
+        return edible_ret_val::failure( _( "We can't eat that.  It's not right for us." ),
+                                        INEDIBLE_MUTATION );
     }
     // Here's why PROBOSCIS is such a negative trait.
     if( has_trait( trait_id( "PROBOSCIS" ) ) && !drinkable ) {
-        return error( INEDIBLE_MUTATION, _( "Ugh, you can't drink that!" ) );
+        return edible_ret_val::failure( _( "Ugh, you can't drink that!" ), INEDIBLE_MUTATION );
     }
 
     if( has_trait( trait_id( "CARNIVORE" ) ) && nutrition_for( food.type ) > 0 &&
         food.has_any_flag( carnivore_blacklist ) && !food.has_flag( "CARNIVORE_OK" ) ) {
-        return error( INEDIBLE_MUTATION, _( "Eww.  Inedible plant stuff!" ) );
+        return edible_ret_val::failure( _( "Eww.  Inedible plant stuff!" ), INEDIBLE_MUTATION );
     }
 
     if( ( has_trait( trait_id( "HERBIVORE" ) ) || has_trait( trait_id( "RUMINANT" ) ) ) &&
         food.has_any_flag( herbivore_blacklist ) ) {
         // Like non-cannibal, but more strict!
-        return error( INEDIBLE_MUTATION, _( "The thought of eating that makes you feel sick." ) );
+        return edible_ret_val::failure( _( "The thought of eating that makes you feel sick." ),
+                                        INEDIBLE_MUTATION );
     }
 
-    return EDIBLE;
+    return edible_ret_val::success();
 }
 
-edible_rating player::will_eat( const item &food, bool interactive ) const
+edible_ret_val player::will_eat( const item &food, bool interactive ) const
 {
-    std::string err;
-    const auto edibility = can_eat( food, &err );
-    if( edibility != EDIBLE ) {
+    const auto ret = can_eat( food );
+    if( !ret ) {
         if( interactive ) {
-            add_msg_if_player( m_info, err.c_str() );
+            add_msg_if_player( m_info, ret.c_str() );
```

Comment:
Shouldn't this be `, "%s", ret.c_str() );` - in case the error messages contains '%' characters?